documentation/diagnosing-slow-builds.md

```diff
@@ -0,0 +1,44 @@
+# Diagnosing slow builds
+
+No build is ever fast enough. It's often worth investing some effort on understanding where MSBuild spends its time when building your code, and thinking about whether it can be improved.
+
+[Obligatory XKCD](https://www.xkcd.com/303/)
+
+Logs at MSBuild’s “normal” level don't capture most of the really interesting information, though.
+
+Try [capturing a binary log](https://aka.ms/msbuild/binlog) in your build; that preserves detailed information and allows for offline analysis. Changing log verbosity levels can affect build performance, but doesn't usually cause dramatic slowdown for full builds. Cross-check the results of with-logging and without-logging builds to make sure they're not completely out of whack before delving deeply into analyzing the binary log.
```

Comment:
```suggestion
Try [capturing a binary log](https://github.com/microsoft/msbuild/blob/master/documentation/wiki/Providing-Binary-Logs.md) in your build. The binary log preserves detailed information and allows for offline analysis. Changing log verbosity levels can affect build performance but doesn't usually slow full builds too dramatically. Cross-check the results of with-logging and without-logging builds to make sure they're reasonable before delving deeply into analyzing the binary log.
```

Reply:
By `Cross-check the results...to make sure they're reasonable`, do you mean making sure the outputs and build times are similar?

Reply:
I'm fine with this suggested change except for going away from the aka.ms link (so long as one of us owns that aka.ms link)

Reply:
I was wrong when I suggested moving away from the aka.ms link. Glancing at my history, it seems I typed `aka.ms/msbuild/buildlog` by accident, got nowhere, and assumed it was a dead link.